civis/tests/test_futures.py

```diff
@@ -86,40 +92,41 @@ def test_check_message_when_job_is_running(self, *mocks):
         self.assertFalse(result._check_message(message))
 
     @mock.patch(api_import_str, return_value=civis_api_spec_base)
-    def test_set_api_result_result_succeeded(self, mock_api):
-        poller = mock.Mock()
-        api_result = mock.Mock()
-        api_result.state = 'succeeded'
+    def test_poller_call_count_poll_on_creation_true(self, mock_api):
+        poller = _create_poller_mock("succeeded")
+        CivisFuture(poller, (1, 2), poll_on_creation=True)
+        assert poller.call_count == 1
 
-        result = CivisFuture(poller, (1, 2))
-        result._set_api_result(api_result)
+    @mock.patch(api_import_str, return_value=civis_api_spec_base)
+    def test_poller_call_count_poll_on_creation_false(self, mock_api):
+        poller = _create_poller_mock("succeeded")
+        CivisFuture(poller, (1, 2), poll_on_creation=False)
         assert poller.call_count == 0
```

Comment:
re: poller count, I've verified that `poll_on_creation=True` (= current default value) does make the poller called, and I've written both True vs False cases here to be explicit.